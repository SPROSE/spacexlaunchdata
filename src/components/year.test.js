import React from 'react';
import { render } from '@testing-library/react';
import Year from './year';
import {shallow} from "enzyme/build/index";

const initialProps = {
        yearsData: {
        year: 0,
        availableYears: [2006, 2007, 2008, 2009],
        showYearsDropdown: false
    }
};

describe('<Year /> with props', () => {
    const container = shallow(<Year {...initialProps} />);
    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });
});

const newProps = {
    yearsData: {
        year: 0,
        availableYears: [2006, 2007, 2008, 2009],
        showYearsDropdown: true
    }
};

test('renders component without dropdown', () => {
    const { getByText } = render(<Year {...newProps} />);
    const dropdownValue = getByText('2008');
    expect(dropdownValue).toBeInTheDocument();
});

describe('<Year /> with dropdown', () => {
    const container = shallow(<Year {...newProps} />);
    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });
});