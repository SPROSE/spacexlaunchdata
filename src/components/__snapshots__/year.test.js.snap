// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Year /> with dropdown should match the snapshot 1`] = `"<div class=\\"relative inline\\"><button class=\\"btn\\">Filter by Year <img src=\\"select.png\\" class=\\"icon\\" alt=\\"select year\\"/></button><div class=\\"dropdownListContainer\\"><ul class=\\"dropdown-list\\"><li class=\\"dropdown-entry\\">2006</li><li class=\\"dropdown-entry\\">2007</li><li class=\\"dropdown-entry\\">2008</li><li class=\\"dropdown-entry\\">2009</li></ul></div></div>"`;

exports[`<Year /> with props should match the snapshot 1`] = `"<div class=\\"relative inline\\"><button class=\\"btn\\">Filter by Year <img src=\\"select.png\\" class=\\"icon\\" alt=\\"select year\\"/></button></div>"`;
